retworkx/__init__.py

```diff
@@ -1817,3 +1817,70 @@ def _digraph_bfs_search(graph, source, visitor):
 @bfs_search.register(PyGraph)
 def _graph_bfs_search(graph, source, visitor):
     return graph_bfs_search(graph, source, visitor)
+
+
+@functools.singledispatch
+def dijkstra_search(graph, source, weight_fn, visitor):
+    """Dijkstra traversal of a graph.
+
+    The pseudo-code for the Dijkstra algorithm is listed below, with the annotated
+    event points, for which the given visitor object will be called with the
+    appropriate method.
```

Comment:
Open to discussion, but having pseudocode that looks more like Python could improve this docstring